src/client/visualizations/base-visualization/highlight.ts

```diff
@@ -15,21 +15,22 @@
  * limitations under the License.
  */
 
-@import '../../imports';
+import { List, Record } from "immutable";
+import { FilterClause } from "../../../common/models/filter-clause/filter-clause";
 
-.hover-multi-bubble {
-  @extend %menu-cont;
-  background: rgba($white, 0.9);
-  position: absolute;
-  padding: 8px;
+export interface HighlightValue {
+  clauses: List<FilterClause>;
+  key: string | null;
+}
 
-  .bucket {
-    color: $text-lighterish;
-    margin-bottom: 6px;
-    white-space: nowrap;
-  }
+const defaultHighlight: HighlightValue = {
+  clauses: List.of(),
+  key: null
+};
+
+export class Highlight extends Record<HighlightValue>(defaultHighlight) {
 
-  .segment-action-buttons {
-    margin-top: 7px;
+  public toString(): string {
+    return `[Highlight ${this.clauses.toString()}]`;
```

Comment:
No key?

Reply:
Honestly, I don't know why we reimplement this method here. Let me check.